### PR TITLE
Fix BodoSQL use in Jupyter (without JIT)

### PR DESCRIPTION
**Pull request by ehsantn**
## Changes included in this PR

Fixes BodoSQL use inside Jupyter without JIT (spawn mode). Updates the quickstart example to not use a JIT function.

## Testing strategy

Tested it manually since we don't have unit test infrastructure for Jupyter yet.

## Checklist
- [ ] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: BodoSQL/bodosql/context.py [hunk 1]
```diff
@@ -1104,6 +1104,9 @@ def sql(self, sql, params_dict=None, dynamic_params_list=None, **jit_options):
             # are not visible to cloudpickle as used by the function. See:
             # test_json_fns.py::test_object_construct_keep_null[no_nested-no_null-with_case]
             if isinstance(dispatcher, SubmitDispatcher):
+                # __builtins__ which is added to glbls by exec causes issues in
+                # Jupyter/IPython pickling.
+                glbls.pop("__builtins__", None)
```
Comment (IsaacWarren) [inline BodoSQL/bodosql/context.py:1109]: Don't we need this to use builtin functions like len though?

**Review by IsaacWarren — APPROVED** [review]
One question otherwise LGTm, thanks Ehsan

File: BodoSQL/bodosql/context.py (hunk 1 above)
Comment (ehsantn) [reply BodoSQL/bodosql/context.py:1109]: Builtin functions are the same on the worker and are available in regular globals. We don't need to pickle and send them.

Comment (IsaacWarren) [reply BodoSQL/bodosql/context.py:1109]: Makes sense, thanks

**Review by scott-routledge2 — APPROVED** [review]

**Pull request merged**